### PR TITLE
OCPBUGS-23248: change Alertmanager form to create using matchers inst…

### PR DESCRIPTION
**Pull request by rhamilto**
…ead of match and match_re

and convert any existing `match` and `match_re` to `matcher` when editing an existing receiver

localhost_9000_monitoring_alertmanagerconfig_receivers_email_edit

### CODE REVIEW

File: frontend/public/components/monitoring/receiver-forms/routing-labels-editor.tsx [hunk 1]
```diff
@@ -72,129 +48,52 @@ export const RoutingLabelEditor = ({ formValues, dispatchFormChange, isDefaultRe
...
         <Trans ns="public">
-          Firing alerts with labels that match all of these selectors will be sent to this receiver.
-          Label values can be matched exactly or with a{' '}
+          Firing alerts with labels that match all of these{' '}
           <ExternalLink
-            href="https://github.com/google/re2/wiki/Syntax"
-            text={t('public~regular expression')}
-          />
-          .
+            href="https://github.com/prometheus/alertmanager/blob/e3fb99cc2d24330fdb81aa1d466c19b3de328b5b/docs/configuration.md#matcher"
```
Comment (jhadvig) [inline frontend/public/components/monitoring/receiver-forms/routing-labels-editor.tsx:58]: Lets rather link to the alerting configuration docs
```suggestion
            href="https://prometheus.io/docs/alerting/latest/configuration/#matcher"
```

**Review by jhadvig — COMMENTED** [review]
One docs nit, otherwise really nice job @rhamilto  👍 🚀

File: frontend/public/components/monitoring/receiver-forms/routing-labels-editor.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/monitoring/receiver-forms/routing-labels-editor.tsx]: Good call.  Updated.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**